sa/model_test.go

```diff
@@ -224,3 +234,86 @@ func TestPopulateAttemptedFieldsBadJSON(t *testing.T) {
 		})
 	}
 }
+
+func TestCerficatesTableContainsDuplicateSerials(t *testing.T) {
+	testCtx := setup(t)
+	defer testCtx.cleanUp()
+
+	serialString := core.SerialToString(big.NewInt(1337))
+
+	// Insert a certificate with a serial of `1337`.
+	testCtx.insertCertificate(t, "1337.com", 1337, 1)
+
+	// This should return the certificate that we just inserted.
+	_, err := SelectCertificate(testCtx.db, serialString)
+	test.AssertNotError(t, err, "received an error for a valid query")
+
+	// Insert a certificate with a serial of `1337` but for a different
+	// hostname.
+	testCtx.insertCertificate(t, "1337.net", 1337, 1)
+
+	// With a duplicate being present, this should error.
+	_, err = SelectCertificate(testCtx.db, serialString)
+	test.AssertError(t, err, "should've received an error for multiple rows")
+}
+
+type testCtx struct {
+	db      *db.WrappedMap
```

Comment:
It looks like the main thing that distinguishes `testCtx` from the return values for `initSA` is that it has a separate `db` element, which you use to call `db.Insert` directly. You could get away with just using `initSA` if you're willing to reach into the returned SA object to get its dbMap, *or* if you're willing to call AddCertificate instead of doing the Insert directly.

Reply:
Oh! Yeah, I misunderstood the previous comment there. Yeah, no need for this connection since the initSA is going to give us one.